core/src/main/java/org/apache/accumulo/core/zookeeper/ZcNode.java

```diff
@@ -83,13 +116,29 @@ private ZcNode() {
    * stat.
    */
   ZcNode(byte[] data, ZcStat zstat, ZcNode existing) {
-    this.data = Objects.requireNonNull(data);
-    this.stat = Objects.requireNonNull(zstat);
```

Comment:
Does not make a difference so feel free to ignore: I think it would be a bit clearer to keep these checks at the top of the constructor instead of at the end.

Reply:
That seems good, moved that to the top.  Also noticed some unused code when making changes and removed that.